xmtp_mls/src/api/identity.rs

```diff
@@ -0,0 +1,245 @@
+use std::collections::HashMap;
+
+use crate::types::InboxId;
+
+use super::{ApiClientWrapper, WrappedApiError};
+use xmtp_id::associations::{DeserializationError, IdentityUpdate};
+use xmtp_proto::{
+    api_client::{XmtpIdentityClient, XmtpMlsClient},
+    xmtp::identity::api::v1::{
+        get_identity_updates_request::Request as GetIdentityUpdatesV2RequestProto,
+        get_identity_updates_response::IdentityUpdateLog,
+        get_inbox_ids_request::Request as GetInboxIdsRequestProto,
+        GetIdentityUpdatesRequest as GetIdentityUpdatesV2Request, GetInboxIdsRequest,
+        PublishIdentityUpdateRequest,
+    },
+};
+
+/// A filter for querying identity updates. `sequence_id` is the starting sequence, and only later updates will be returned.
+pub struct GetIdentityUpdatesV2Filter {
+    pub inbox_id: String,
+    pub sequence_id: Option<u64>,
+}
+
+impl From<GetIdentityUpdatesV2Filter> for GetIdentityUpdatesV2RequestProto {
+    fn from(filter: GetIdentityUpdatesV2Filter) -> Self {
+        Self {
+            inbox_id: filter.inbox_id,
+            sequence_id: filter.sequence_id.unwrap_or(0),
+        }
+    }
+}
+
+pub struct IdentityUpdateV2 {
```

Comment:
Could this be `IdentityUpdateV3` since it's for XMTP v3? Apparently it's second version of IdentityUpdate but I think it'd be more clear to unify the versioning across the board.

Reply:
You're right, it's definitely confusing. We just already had a struct with `IdentityUpdate`. Maybe I can rename to `InboxUpdate`

Reply:
The good news is that once we're done this transition we can delete the old `IdentityUpdate` and rename everything

Reply:
`InboxUpdate` sounds great